src/puppetlabs/puppetdb/utils.clj

```diff
@@ -343,6 +360,29 @@
     {:command command :version (Integer. version)
      :certname certname :payload (json/->RawJsonString payload)}))
 
+(def ^Charset utf8 StandardCharsets/UTF_8)
+
+(defn utf8-bytes [s] (.getBytes ^String s utf8))
+
+(defn utf8-length
+  "Return the length in bytes of the given string when encoded in UTF-8"
+  [s]
+  (count (.getBytes ^String s utf8)))
+
+(defn utf8-truncate
+  "Truncate the given string such that its UTF-8 representation is at most
+  `max-bytes` long. Note that the returned string may be empty."
+  [s max-bytes]
+  (let [utf8-bytebuff (ByteBuffer/allocate max-bytes)
```

Comment:
TIL about CharBuffer/wrap and the UTF8 encoder class. :+1: 

One question I have is, should be be truncating the string to n chars _before_ we truncate? I'm just thinking of how this encoder/flush stuff is working. If we have a string that is too long, it could hit that `max-bytes` limit mid char, or even if not mid-char, the byte buffer would be full and flush-result would have no effect, is that right?